pkg/tests/ossm/bug_istiopods_test.go

```diff
@@ -56,6 +60,72 @@ func TestIstiodPodFailsAfterRestarts(t *testing.T) {
 	})
 }
 
+func TestControllerFailsToUpdatePod(t *testing.T) {
+	NewTest(t).Groups(Full, Disconnected, ARM).Run(func(t TestHelper) {
+		t.Log("Verify that the controller does not fails to update the pod when the member controller couldn't add the member-of label")
+		t.Log("References: \n- https://issues.redhat.com/browse/OSSM-2169\n- https://issues.redhat.com/browse/OSSM-2420")
+
+		// Add timestamp to the namespaces to avoid conflicts during tests execution
+		ti := time.Now()
+		nameTemplate := fmt.Sprintf("test-smcp-%d%d-%02d%02d-",
+			env.GetSMCPVersion().Major,
+			env.GetSMCPVersion().Minor,
+			ti.Hour(), ti.Minute())
+
+		namespaces := util.GenerateStrings(nameTemplate, 100)
+		re := regexp.MustCompile(fmt.Sprintf(`error adding member-of label to namespace (%s\d+)`, nameTemplate))
+
+		t.Cleanup(func() {
+			oc.DeleteNamespace(t, namespaces...)
+			oc.ApplyString(t, meshNamespace, smmr)
+		})
+
+		DeployControlPlane(t)
+
+		t.LogStep("Add namespaces to the SMMR")
+		oc.ApplyString(t, meshNamespace, createSMMRManifest(namespaces...))
+
+		istioOperatorPodName := oc.GetAllResoucesNamesByLabel(t, "openshift-operators", "pod", "name=istio-operator")[0]
+
+		// Initially assumed 1 iteration, however, as issue can be flaky it could be increased up to 5 iterations
+		count := 2
+		for i := 1; i < count; i++ {
+			t.LogStepf("Create/Recreate 100 Namespaces, attempt #%d", i)
+			oc.RecreateNamespace(t, namespaces...)
+
+			t.LogStepf("Check istio-operator logs for 'Error updating pod's labels', attempt #%d", i)
+			output := shell.Execute(t,
+				fmt.Sprintf("oc logs %s -n openshift-operators", istioOperatorPodName),
+				assert.OutputDoesNotContain(
+					"Error updating pod's labels",
+					"Found no updating pod's labels error",
+					"Expected to find no error updating pod's labels, but got",
+				))
+
+			t.LogStepf("Check istio-operator logs for 'error adding member-of label' errors, attempt #%d", i)
+			matches := re.FindStringSubmatch(output)
+			if len(matches) > 1 {
+				namespaceName := matches[1]
+				successMessage := fmt.Sprintf(`Added member-of label to namespace","ServiceMeshMember":"%s/default","namespace":"%s`, namespaceName, namespaceName)
+				if strings.Contains(output, successMessage) {
+					t.LogSuccessf("Found error and success message for namespace %s: %s", namespaceName, successMessage)
+					break
+				} else {
+					t.Log(output)
+					t.Fatalf("Was not found success message after error for namespace %s: %s", namespaceName, successMessage)
+				}
+			} else {
+				if count < 6 {
+					count++
+					t.Logf("Was not found any 'error adding member-of label' error, repeat (max 5), attempt #%d", i)
+				} else {
+					t.Logf("Was not found any 'error adding member-of label' error, stop test, attempt #%d", i)
```

Comment:
`stop test` , shouldn't be there t.Fail or something?

Reply:
I decided to make just successful stop after 5 attempts to face error.

I found out one new issue with latest mtt run so will try to rework that part too

Reply:
As it is flaky conflict that previously led to error... So I do not know should we spent extra time to always trying to catch that conflict :thinking: